src/scales/scale.time.js

```diff
@@ -360,8 +360,9 @@ module.exports = function(Chart) {
 		getLabelForIndex: function(index, datasetIndex) {
 			var me = this;
 			var label = me.chart.data.labels && index < me.chart.data.labels.length ? me.chart.data.labels[index] : '';
+			var firstData = me.chart.data.datasets[datasetIndex].data[index];
 
-			if (typeof me.chart.data.datasets[datasetIndex].data[0] === 'object') {
+			if (firstData !== null && typeof firstData === 'object') {
 				label = me.getRightValue(me.chart.data.datasets[datasetIndex].data[index]);
```

Comment:
Since it's not the first data anymore, `value` could be a better name than `firstData` and you could use it here as well and save a few bits:

```javascript
if (value !== null && typeof value === 'object') {
    label = me.getRightValue(value);
}
```